aten/src/ATen/native/native_functions.yaml

```diff
@@ -646,13 +646,30 @@
   dispatch:
     CPU: bmm_cpu
     CUDA: bmm_cuda
+    SparseCPU: bmm_sparse_cpu
+    SparseCUDA: bmm_sparse_cuda
+  supports_named_tensor: True
+
+- func: _bmm(Tensor self, Tensor mat2, bool deterministic) -> Tensor
```

Comment:
In the issue we filed, we concluded that the kwargs didn't work because you didn't update the out variant too when you did it. So let's try kwarging this again.

Reply:
Oh I think I misunderstood that. I'll try that.

Reply:
What is the proper way to make the out variant use TensorOptions? I tried this:
```
- func: bmm.out(Tensor self, Tensor mat2, *, bool? deterministic=False, Tensor(a!) out) -> Tensor(a!)
```
But that apparently just turns deterministic into an `optional<bool>` for the out variant, even though it's a TensorOptions in the regular variant. When I try to build, I get this:
```
../aten/src/ATen/native/LinearAlgebra.cpp:313:61: error: could not convert 'options' from 'c10::TensorOptions' to 'c10::optional<bool>'
   return at::native::bmm_out_cpu(result, self, mat2, options);
```
I searched for other existing out functions that use TensorOptions, and I don't think there are any: `grep -R TensorOptions aten/src/ATen/native | grep '_out'`

Reply:
Why do you want the out variant to use TensorOptions? Thi sisn't a factory function and I don't feel like that should happen here. Can you post the entire diff you did, and the auto-generated code in that case?

Reply:
Apparently I still don't understand what I need to do. I thought that in order to make the kwargs work, I need to make these changes in native_functions.yaml:
```
- func: bmm(Tensor self, Tensor mat2, *, bool? deterministic=False) -> Tensor
...
- func: bmm.out(Tensor self, Tensor mat2, *, bool? deterministic=False, Tensor(a!) out) -> Tensor(a!)
```

After that, I thought I just needed to add a TensorOptions argument to the end of all the bmm and bmm_out function implementations.

What do I need to do instead?

Reply:
Ok it seems like I needed to add optional<bool> to all of the bmm function implementations. That seems to work. I'm not sure why I thought I needed TensorOptions.

Reply:
Great!